Universal/Tests/CodeAnalysis/MixedBooleanOperatorUnitTest.inc

```diff
@@ -0,0 +1,41 @@
+<?php
+
+if (true && true || true);
+if ((true && true) || true);
+if (true && (true || true));
+
+$var = true && true || true;
+$var = (true && true) || true;
+$var = true && (true || true);
+
+$complex = true && (true || true) && true;
+$complex = true && (true || true) || true;
+
+if (
+    true
+    && true
+    || true
+);
+
+if (
+    true
+    && (
+        true
+        || true
+    )
+);
+
+if (true && foo(true || true));
+if (true && foo(true && true || true));
+if (true && $foo[true || true]);
+if (true && $foo[true && true || true]);
+
+if (true && foo(true) || true);
+if (true && $foo[true] || true);
+if (true && foo($foo[true]) || true);
```

Comment:
I can add some to make sure, but other control structures are not really necessary to test, because the “working space” of the Sniff is limited to a single expression and the stuff that is legal to put into an expression is clearly delineated by either the semicolon or the various parentheses / brackets.

Reply:
In which case, I suggest adding a test with something along the lines of. the below and to see if your statement still holds up....:
```php
for ( $i = true && false, $j = false || true; $i && $j || $k, $i < $j; $i++ ) {}
```

Oh and some tests with ternaries would also not go amiss....